src/ucp/rndv/rndv_put.c

```diff
@@ -414,6 +421,22 @@ ucp_proto_rndv_put_zcopy_query(const ucp_proto_query_params_t *params,
                       UCP_PROTO_ZCOPY_DESC, put_desc);
 }
 
+static ucs_status_t ucp_proto_rndv_put_zcopy_reset(ucp_request_t *request)
+{
+    const ucp_proto_rndv_put_priv_t *rpriv = request->send.proto_config->priv;
+    ucp_lane_map_t acked_map               = rpriv->atp_map;
+    size_t acked_size                      = 0;
+
+    while (acked_map != request->send.rndv.put.atp_map) {
```

Comment:
then maybe something like:
```
acked_map = rpriv->atp_map & ~(rpriv->atp_map & request->send.rndv.put.atp_map);
ucs_for_each_bit(i, acked_map) {
    acked_size += ucp_proto_rndv_put_common_ack_size(request, acked_map);
    acked_map &= ~UCS_BIT(i);
}
```

Reply:
will not work because `acked_map` should start from full `atp_map`

Reply:
so can change `acked_map` initial value?

Reply:
so how can you stop iteration when you reach acked_map == atp_map?

Reply:
AFAIU it's possible to calculate all required bits in advance and do the loop thru all of them

Reply:
we can compare request->send.rndv.put.atp_map to 0
if they are different, it means we did not ack all lanes yet, so ack_size is popcount(rpriv->atp_map) - 1
otherwise, (atp_map is 0) we acked all lanes, so offset is dt_iter length